### PR TITLE
Add new `map` options to `px` and update plotly.js/master at 056799dfc705a4533935ae3169cc94d35bc44830

### PR DESCRIPTION
**Pull request by archmoj**
Related to https://github.com/plotly/plotly.js/pull/7015.

### CODE REVIEW

**Comment by archmoj** [comment]
cc: [birkskyum](https://github.com/birkskyum)

File: packages/python/plotly/plotly/express/_doc.py [hunk 1]
```diff
@@ -531,6 +531,13 @@
         "Dict keys are `'lat'` and `'lon'`",
         "Sets the center point of the map.",
     ],
+    map_style=[
+        "str (default `'basic'`, needs Map API token)",
+        "Identifier of base map style, some of which require a Map or Stadia Maps API token to be set using `plotly.express.set_map_access_token()`.",
```
Comment (archmoj) [inline packages/python/plotly/plotly/express/_doc.py:536]: Update description:
None of these require a token.

File: packages/python/plotly/plotly/express/_doc.py [hunk 2]
```diff
@@ -531,6 +531,13 @@
...
+        "Allowed values which do not require a token are `'open-street-map'`, `'white-bg'`, `'carto-positron'`, `'carto-darkmatter'`.",
+        "Allowed values which require a Map API token are `'basic'`, `'streets'`, `'outdoors'`, `'light'`, `'dark'`, `'satellite'`, `'satellite-streets'`.",
+        "Allowed values which require a Stadia Maps API token are `'stamen-terrain'`, `'stamen-toner'`, `'stamen-watercolor'`.",
+    ],
```
[the remark this answers is not in the document]
Comment (LiamConnors) [reply packages/python/plotly/plotly/express/_doc.py:540]: @emilykl @archmoj , does this look correct to you? I understand we took out the stamen values

[the remark this answers is not in the document]
Comment (archmoj) [reply packages/python/plotly/plotly/express/_doc.py:540]: We took out `stamen` in https://github.com/plotly/plotly.js/pull/7085.
But this list is not complete. Here is the list: 
```
'basic'
'carto-darkmatter'
'carto-darkmatter-nolabels'
'carto-positron'
'carto-positron-nolabels'
'carto-voyager'
'carto-voyager-nolabels'
'dark'
'light'
'open-street-map'
'outdoors'
'satellite'
'satellite-streets'
'streets'
'white-bg'
```

**Comment by LiamConnors** [comment]
This looks good to me. Was there a particular example you couldn't get to render @archmoj

**Review by ndrezn — COMMENTED** [review]
Is it possible to add a deprecation warning similar to ,https://github.com/plotly/plotly.js/pull/7087, but in this case have it appear in the Python console _in addition to_ in the browser console?

**Comment by archmoj** [comment]
> Is it possible to add a deprecation warning for Mapbox traces similar to [plotly/plotly.js#7087](https://github.com/plotly/plotly.js/pull/7087), but in this case have it appear in the Python console _in addition to_ in the browser console?

If possible, that should be done in a separate PR. 
This PR is already huge.

File: packages/python/plotly/plotly/express/_doc.py (hunk 1 above)
Comment (archmoj) [reply packages/python/plotly/plotly/express/_doc.py]: @LiamConnors Would you please edit this description?
Thank you!

Comment (archmoj) [reply packages/python/plotly/plotly/express/_doc.py]: I adjusted these in 7e2eefcc340a95b65c3fe433f56adb7fbe739b4a.
Resolving...

**Comment by ndrezn** [comment]
@archmoj makes sense to me -- opened #4730 to cover that case.

**Comment by archmoj** [comment]
@LiamConnors Please take over this PR and once ready merge it into your doc PR or master?
I thought you may want to revert big commit of plotly.js update [ba9c413](https://github.com/plotly/plotly.py/pull/4726/commits/ba9c413ced84eb5e0a2abef1df5a310971fbf47e) and just add the px parts to have viewable diffs. No?

**Review by gvwilson — APPROVED** [review]

**Pull request merged**